src/app/profiles/profiles.editor.modal.ts

```diff
@@ -0,0 +1,583 @@
+                </form>
+           </div>
+       </mat-card-content>
+       <mat-card-actions style="padding: 0px 25px">
+           <button mat-raised-button (click)="save()" color="primary">SAVE</button>
+           <button mat-raised-button (click)="close()" color="warn">CANCEL</button>
+       </mat-card-actions>
+       <mat-card-footer style="padding: 0px 25px">
+           <mat-progress-bar *ngIf="is_busy" mode="indeterminate"></mat-progress-bar>
+           <div
+               *ngIf="data.profile.reference_type == 'preference' && data.profile.reference.status === 1"
+               >Email not validated. Check your email or <a href="javascript:void(0)" (click)="resend_validate_email()">re-send</a>.
+           </div>
+       </mat-card-footer>
+   </mat-card>
+    `
+})
+
+export class ProfilesEditorModal {
+    @Input() value: any[];
+    field_errors;
+    allowed_domains = [];
+    is_valid = false;
+    aliases_unique = [];
+    is_busy = false;
+    is_delete =false;
+    is_update = false;
+    is_create = false;
+    is_create_main = false;
+    type;
+    is_visible_smtp_detail = false;
+    is_different_reply_to = false;
+    localpart;
+    constructor(
+        public rmm: RMM,
+        private http: HttpClient,
+        public snackBar: MatSnackBar,
+        public dialog_ref: MatDialogRef<ProfilesEditorModal>,
+        @Inject(MAT_DIALOG_DATA) public data: any
+    ) {
+        if ( data && data.type ) {
+            this.type = data.type;
+            delete data.type;
+        }
+        if ( data.profile && data.profile.email ) {
+            this.set_localpart(data);
+        }
+        if ( ! data || ! Object.keys(data).length || !data.profile ) {
+            data = { profile : { } };
+            let self = this;
+            data.profile.name = ['first_name','last_name'].map((attr) => {
+                console.log('attr', attr, self.rmm.me.data[attr]);
+                return self.rmm.me.data[attr]
+            }).join(' ')
+        }
+        this.data = data;
+        this.check_reply_to(this.data);
+    }
+    check_reply_to(data) {
+        if (data && data.profile.email && data.profile.reply_to && 
+            data.profile.reply_to !== data.profile.email) {
+            this.is_different_reply_to=true;
+            return;
+        }
+        this.is_different_reply_to=false;
+    }
+    set_localpart(data) {
+        if (data.profile.email.match(/@/g)) {
+            this.localpart = data.profile.email.replace(/@.+/g, '');
+            var regex = /(.+)@(.+)/g;
+            var match = regex.exec(data.profile.email);
+            data.profile.preferred_runbox_domain = match[2];
+        } else {
+            this.localpart = data.profile.email;
+            data.profile.preferred_runbox_domain = this.localpart;
+        }
+    }
+    save() {
+        if ( this.is_create || this.is_create_main ) { this.create() }
+        else { this.update() }
+    }
+    create(){
+        this.is_busy = true;
```

Comment:
Remember to set this to false not only when the HTTP query succeeds, but also if it fails. Right now it's only set back to false if the HTTP query didn't fail, and the REST response is `success`.